caf-audit-monkey/src/main/java/com/github/cafaudit/auditmonkey/MonkeyConfig.java

```diff
@@ -0,0 +1,438 @@
+        }
+        
+        correlationId = System.getProperty(MonkeyConstants.CAF_AUDIT_CORRELATION_ID, System.getenv(MonkeyConstants.CAF_AUDIT_CORRELATION_ID));
+        if(null == correlationId || correlationId.isEmpty()) {
+            correlationId = UUID.randomUUID().toString();
+            LOG.info("No [" + MonkeyConstants.CAF_AUDIT_CORRELATION_ID + "] supplied defaulting to [" + correlationId + "]");
+        }
+        
+        userId = System.getProperty(MonkeyConstants.CAF_AUDIT_USER_ID, System.getenv(MonkeyConstants.CAF_AUDIT_USER_ID));
+        if(null == userId || userId.isEmpty()) {
+            userId = "road.runner@acme.com";
+            LOG.info("No [" + MonkeyConstants.CAF_AUDIT_USER_ID + "] supplied defaulting to [" + userId + "]");
+        }
+        
+        /*
+         * Monkey
+         */
+        monkeyMode = System.getProperty(MonkeyConstants.CAF_AUDIT_MONKEY_MODE, System.getenv(MonkeyConstants.CAF_AUDIT_MONKEY_MODE));
+        if(null == monkeyMode || monkeyMode.isEmpty()) {
+            monkeyMode = MonkeyConstants.CAF_AUDIT_STANDARD_MONKEY;
+            LOG.info("No [" + MonkeyConstants.CAF_AUDIT_MONKEY_MODE + "] supplied defaulting to [" + monkeyMode + "]");
+        } 
+        else if(!monkeyMode.equalsIgnoreCase(MonkeyConstants.CAF_AUDIT_STANDARD_MONKEY) && !monkeyMode.equalsIgnoreCase(MonkeyConstants.CAF_AUDIT_RANDOM_MONKEY)) {
+            String errorMsg = "The " + MonkeyConstants.CAF_AUDIT_MONKEY_MODE + " supplied [" + monkeyMode + "] does not match the available modes [" + MonkeyConstants.CAF_AUDIT_STANDARD_MONKEY + ", " + MonkeyConstants.CAF_AUDIT_RANDOM_MONKEY + "]"; 
+            LOG.error(errorMsg);
+            throw new RuntimeException(errorMsg);
+        }
+
+        String numOfEventsStr = System.getProperty(MonkeyConstants.CAF_AUDIT_MONKEY_NUM_OF_EVENTS, System.getenv(MonkeyConstants.CAF_AUDIT_MONKEY_NUM_OF_EVENTS));
+        if(null == numOfEventsStr || numOfEventsStr.isEmpty()) {
+            numOfEvents = 1; 
+            LOG.warn("No [" + MonkeyConstants.CAF_AUDIT_MONKEY_NUM_OF_EVENTS + "] supplied defaulting to [" + numOfEvents + "]");
+        }
+        else {
+            numOfEvents = Integer.parseInt(numOfEventsStr);
+            if(numOfEvents <= 0) {
+                numOfEvents = 1;
+                LOG.warn("Invalid [" + MonkeyConstants.CAF_AUDIT_MONKEY_NUM_OF_EVENTS + "] supplied defaulting to [" + numOfEvents + "]");
+            }
+        }
+        
+        String numOfThreadsStr = System.getProperty(MonkeyConstants.CAF_AUDIT_MONKEY_NUM_OF_THREADS, System.getenv(MonkeyConstants.CAF_AUDIT_MONKEY_NUM_OF_THREADS));
+        if(null == numOfThreadsStr || numOfThreadsStr.isEmpty()) {
+            numOfThreads = 1; 
+            LOG.info("No [" + MonkeyConstants.CAF_AUDIT_MONKEY_NUM_OF_THREADS + "] supplied defaulting to [" + numOfThreads + "]");
+        }
+        else {
+            numOfThreads = Integer.parseInt(numOfThreadsStr);
```

Comment:
This is not a customer facing item so I am not too worried.  If it was customer facing additional validation may indeed be required.

If someone provides a value of  say [2.75] threads, then so be it, they are too stupid to use the Monkey!  :-P